ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_tensor.py

```diff
@@ -1916,6 +1916,42 @@ def test_paddle_tensor_isnan(
     )
 
 
+# lerp
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="lerp",
+    dtypes_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"), num_arrays=3, shared_dtype=True
+    ),
+)
+def test_paddle_tensor_lerp(
+    dtypes_and_x,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+    backend_fw,
+):
+    input_dtype, x = dtypes_and_x
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        backend_to_test=backend_fw,
+        init_all_as_kwargs_np={"data": x[0]},
+        method_input_dtypes=input_dtype,
+        method_all_as_kwargs_np={
+            "y": x[1],
+            "weight": x[2],
```

Comment:
We're supposed to test the `weights` argument as both `float` or `Tensor` since the native framework supports both.

Reply:
@he11owthere This generates an array input. We also need to test this using `st.floats`

Reply:
@hmahmood24, I have implemented the necessary modifications, allowing the `weight` parameter to be tested for both zero-dimensional `float` values and arrays containing `float` values. 
Thank you very much for your guidance.

Reply:
> @hmahmood24, I have implemented the necessary modifications, allowing the `weight` parameter to be tested for both zero-dimensional `float` values and arrays containing `float` values. Thank you very much for your guidance.

@he11owthere Unfortunately, this will not do because in the end a 0-dimensional array is simply `np.array(2)` but this is not really a `float` type is what I'm talking about. You need to combine two strategies together: one for generating an array input and the other for generating floats. Look up `st.composite` for creating a custom strategy function that sometimes spits out array values and sometimes spits out simply floats.

Reply:
@hmahmood24, I apologize for any confusion caused. I've now implemented a custom strategy function, `_get_dtype_and_values_for_lerp`, that generates the required values for the `weight` parameter. 

Thank you for explaining everything! 😊